### PR TITLE
Logging_debug

### PR DESCRIPTION
**Pull request by Ravishdeep10**
Added logging so now when debugging instead of printing, debug messages shld now appear in a file named debug.log. Other than that added a parameter drop down for parameters which are Simobjects.

### CODE REVIEW

File: m5_calls.py [hunk 1]
```diff
@@ -201,7 +199,7 @@ def traverse_hierarchy_root(sym_catalog, symroot):
         name, simroot = traverse_params(sym_catalog, symroot, root)
         name, simroot = traverse_ports(sym_catalog, symroot, simroot)
     except:
-        logging.error("Could not create simobject tree")
+         logging.error("Could not create simobject tree")
```
Comment (rohitdhamankar) [inline m5_calls.py:202]: would this space cause syntax error

Comment (Ravishdeep10) [reply m5_calls.py:202]: Nice catch !

File: m5_calls.py [hunk 2]
```diff
@@ -190,14 +194,24 @@ def traverse_ports(sym_catalog, symobject, simobject):
...
         instantiated objs"""
-    root = symroot.sim_object_instance
-    name, simroot = traverse_params(sym_catalog, symroot, root)
-    name, simroot = traverse_ports(sym_catalog, symroot, simroot)
+    try:
+        root = symroot.sim_object_instance
+        name, simroot = traverse_params(sym_catalog, symroot, root)
+        name, simroot = traverse_ports(sym_catalog, symroot, simroot)
+    except:
+        logging.error("Could not create simobject tree")
```
Comment (powerjg) [inline m5_calls.py:202]: You probably want to include the exception in the log. This way you will know *why* the simobject tree couldn't be created :)

Also, you should probably re-raise the exception into the GUI logic so you can pop up a window to tell the user that something went wrong.

File: m5_calls.py [hunk 3]
```diff
@@ -190,14 +194,24 @@ def traverse_ports(sym_catalog, symobject, simobject):
...
     return symroot.name, simroot
 
 def instantiate_model():
-    m5.instantiate()
+    try:
+        m5.instantiate()
+    except AttributeError:
+        logging.error("Instantiate error on proxy param")
```
Comment (powerjg) [inline m5_calls.py:209]: Ditto

File: m5_calls.py [hunk 4]
```diff
@@ -190,14 +194,24 @@ def traverse_ports(sym_catalog, symobject, simobject):
...
 
 def simulate():
-    exit_event = m5.simulate()
-    print('Exiting @ tick %i because %s' %(m5.curTick(), exit_event.getCause()))
+    try:
+        exit_event = m5.simulate()
+        print('Exiting @ tick %i because %s' %(m5.curTick(), \
+            exit_event.getCause()))
+    except AttributeError:
+        logging.error("Simulation error")
```
Comment (powerjg) [inline m5_calls.py:217]: Ditto

**Pull request merged**